### PR TITLE
 Temporary directory for tests [v3]

### PR DESCRIPTION
**Pull request by apahim**
v3:
- Improved docs.
- Improved selftests.
- Cleanup on Avocado crash.

v2: #1674 
- Improved selftests.
- Documentation.
- Renamed environment variable.
- Unset the env var on plugin/post.

v1: #1673 
- Job pre/post plugin to create/destroy the directory and expose its path via environment variable.
- Make the directory available via `Test()` API.

### CODE REVIEW

File: docs/source/WritingTests.rst [hunk 1]
```diff
@@ -1190,29 +1190,34 @@ tests can do and also they can modify the test parameters.
...
+| AVOCADO_TEST_SRCDIR         | Source directory for the test         | /var/tmp/avocado_Bjr_rd/my-test.sh/src                                                              |
++-----------------------------+---------------------------------------+-----------------------------------------------------------------------------------------------------+
+| AVOCADO_TESTS_COMMON_TMPDIR | Temporary directory created by the    | /var/tmp/avocado_XhEdo/                                                                             |
+|                             | `teststmpdir` plugin. The directory   |                                                                                                     |
+|                             | is persistent throughout the tests    |                                                                                                     |
+|                             | in the same Job                       |                                                                                                     |
```
Comment (ldoktor) [inline docs/source/WritingTests.rst]: Yep, I like this longer version...

File: selftests/functional/test_teststmpdir.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+INSTRUMENTED_SCRIPT = """import os
+import tempfile
+from avocado import Test
+class MyTest(Test):
+    def test1(self):
+        file = os.path.join(self.teststmpdir,
+                            next(tempfile._get_candidate_names()))
+        open(file, "w+").close()
+        path, dirs, files = os.walk(self.teststmpdir).next()
```
Comment (apahim) [inline selftests/functional/test_teststmpdir.py]: None. Just missed your comment.

**Comment by apahim** [comment]
@ldoktor, updated.

**Comment by clebergnu** [comment]
[Per-job tmpdir for direct use from tests](https://trello.com/c/P1JOtVY7/863-per-job-tmpdir-for-direct-use-from-tests)

**Pull request merged**